e2e/testcases/reconciler_manager_test.go

```diff
@@ -603,3 +603,38 @@ func TestAutopilotReconcilerAdjustment(t *testing.T) {
 	})
 
 }
+
+func TestAllowVerticalScaleResourceAdjustment(t *testing.T) {
+	nt := nomostest.New(t, ntopts.SkipMonoRepo)
+
+	reconcilerDeployment := &appsv1.Deployment{}
+	if err := nt.Validate(nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem, reconcilerDeployment); err != nil {
+		nt.T.Fatal(err)
+	}
+	rs := &v1beta1.RootSync{}
+	if err := nt.Get(configsync.RootSyncName, configsync.ControllerNamespace, rs); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// allow vertical scale
+	_, err := nt.Kubectl("patch", "deployment", "reconciler-manager", "-n", "config-management-system", "-p", "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"args\":[\"--enable-leader-election\", \"--allow-vertical-scale\"],\"name\":\"reconciler-manager\"}]}}}}")
+	if err != nil {
+		nt.T.Fatalf("Error enabling vertical scale in deployment", err)
+	}
+
+	// update numbers are compliant with autopilot requirement
+	var updatedFirstContainerCPURequest int64 = 430
+	var updatedFirstContainerMemoryRequest int64 = 200
+
+	// simulating VPA update to deployment
+	_, err = nt.Kubectl("patch", "deployment", "root-reconciler", "-n", "config-management-system", "-p", fmt.Sprintf("{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"%s\",\"resources\":{\"requests\":{\"cpu\":\"%dm\",\"memory\":\"%dMi\"}}}]}}}}", "hydration-controller", updatedFirstContainerCPURequest, updatedFirstContainerMemoryRequest))
+	if err != nil {
+		nt.T.Fatalf("Error simulating VPA deployment update", err)
+	}
+
+	nomostest.Wait(nt.T, "Waiting for patch deployment update", nt.DefaultWaitTimeout, func() error {
```

Comment:
This is not waiting for patch deployment update. This is to confirm the reconciler-manager doesn't revert the change.
I would suggest you update another field of the Deployment, and validate that field can be reverted by the reconciler-manager, while the "VPA adjusted resources" remain the same.

Reply:
updated comments for clarity.

added minReadySeconds update to the test and verify it getting reverted